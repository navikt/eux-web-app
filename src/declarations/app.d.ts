import { OptionTypeBase } from 'react-select'

export interface Option extends OptionTypeBase {
  label: string
  navn?: string
  value: string
}

export type Options = Array<Option>
export type Loading = {[key: string]: boolean}
export type Params = {[k: string] : string}
export interface ParamPayload {
  key: string,
  value?: any
}
