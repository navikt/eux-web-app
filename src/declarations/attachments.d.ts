import { JoarkBrowserMode, JoarkBrowserType } from 'applications/Vedlegg/JoarkBrowser'
import { Context, Item } from '@navikt/tabell'

export interface JoarkFileVariant {
  variantformat: string
  filnavn: string | null
}

export interface JoarkDoc {
  dokumentInfoId: string
  tittel?: string | null | undefined
  dokumentvarianter: Array<JoarkFileVariant>
}

export interface JoarkPoster {
  journalpostId: string
  tittel: string
  tema: string
  datoOpprettet: string
  tilleggsopplysninger?: Array<any> | null
  dokumenter: Array<JoarkDoc>
}

export type JoarkType = 'joark'

export interface JoarkFile {
  type: JoarkType

  journalpostId: string
  dokumentInfoId: string
  variant: JoarkFileVariant | undefined

  title: string | null | undefined
  tema: string
  date: Date
}

interface Content {
  base64: string
}

export interface JoarkPreview {
  fileName: string | undefined
  contentType: string
  filInnhold: string
}

export interface JoarkBrowserItem extends Item {
  hasSubrows: boolean
  type: JoarkBrowserType

  journalpostId: string
  dokumentInfoId: string | undefined
  variant: JoarkFileVariant | undefined

  title: string
  tema: string | undefined
  date: Date
}

export type JoarkBrowserItems = Array<JoarkBrowserItem>

export interface JoarkBrowserItemWithContent extends JoarkBrowserItem {
  content: Content
  name: string
  size: number
  mimetype: string
}

export interface JoarkBrowserContext extends Context {
  existingItems: JoarkBrowserItems
  gettingJoarkFile: boolean
  clickedPreviewItem: JoarkBrowserItem | undefined,
  mode: JoarkBrowserMode
}

export interface JoarkList {
  data: {
    dokumentoversiktBruker: {
      journalposter: Array<JoarkPoster>
    }
  }
}

export interface SEDAttachmentPayload {
  fnr?: string
  rinaId?: string
  rinaDokumentId?: string
}

export interface SEDAttachmentPayloadWithFile extends SEDAttachmentPayload {
  journalpostId: string
  dokumentInfoId: string
  variantformat: string
  filnavn: string
}

export interface SavingAttachmentsJob {
  total: JoarkBrowserItems
  saved: JoarkBrowserItems
  saving: JoarkBrowserItem | undefined
  remaining: JoarkBrowserItems
}
