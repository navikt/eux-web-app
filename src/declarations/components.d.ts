export interface ModalButton {
  onClick?: () => void;
  disabled ?: boolean;
  main?: boolean;
  text: string;
}

export interface ModalContent {
  modalTitle?: string | null;
  modalContent ?: JSX.Element |string |null;
  modalText ?: string |null;
  modalButtons?: Array<ModalButton> |null;
  closeButton?: boolean | null;
}

export type AllowedLocaleString = 'en' | 'nb'
