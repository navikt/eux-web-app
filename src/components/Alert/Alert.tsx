import FilledRemoveCircle from 'assets/icons/filled-version-remove-circle'
import classNames from 'classnames'
import _ from 'lodash'
import AlertStripe, { AlertStripeType } from 'nav-frontend-alertstriper'
import PT from 'prop-types'
import React from 'react'
import styled, { keyframes } from 'styled-components'

export type AlertStatus = 'OK' | 'ERROR' | 'WARNING'

type AlertStatusClasses = {[status in AlertStatus]: AlertStripeType}

type AlertType = 'client' | 'server'

interface AlertError {
  status?: AlertStatus | undefined
  message ?: JSX.Element | string
  error?: string | undefined
  uuid ?: string | undefined
}

export interface AlertProps {
  className ?: string
  error?: AlertError | string
  fixed?: boolean
  message?: JSX.Element | string
  onClose?: () => void
  status?: AlertStatus
  type?: AlertType
}

/* const AlertErrorPropType = PT.shape({
  status: PT.string,
  message: PT.string,
  error: PT.string,
  uuid: PT.string
}) */

export const errorTypes: AlertStatusClasses = {
  OK: 'suksess',
  ERROR: 'feil',
  WARNING: 'advarsel'
}

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`
const AlertDiv = styled(AlertStripe)`
  opacity: 0;
  animation: ${fadeIn} 0.5s forwards;
  position: relative;

  .alertstripe__tekst {
     flex: auto !important; /* because IE11 */
     margin: auto 0;
     max-width: 100vw;
     display: flex;
     justify-content: space-between;
  }
  .fixed {
    position: fixed;
    top: 0.25rem;
    left: 50%;
    transform: translateX(-50%);
    z-index: 10;
  }
  .closeIcon {
    position: absolute;
    top: 0.25rem;
    right: 0.25rem;
    cursor: pointer;
  }
  .type-server {
    border-radius: 0px !important;
    border: 0px !important;
    .alertstripe__tekst {
      display: flex !important;
      justify-content: space-between;
      max-width: none !important;
    }
  }
`

export const Alert: React.FC<AlertProps> = ({
  className, error, fixed, message, onClose, status = 'ERROR', type
}: AlertProps): JSX.Element => {
  let _message: JSX.Element | string | undefined = message

  const onCloseIconClicked = (): void => {
    if (_.isFunction(onClose)) {
      onClose()
    }
  }

  const printError = (error: AlertError | string): string => {
    const errorMessage: Array<JSX.Element | string> = []
    if (_.isString(error)) {
      return error
    }
    /* if (error.status) {
      errorMessage.push(error.status)
    } */
    if (error.message) {
      errorMessage.push(error.message)
    }
    if (error.error) {
      errorMessage.push(error.error)
    }
    if (error.uuid) {
      errorMessage.push(error.uuid)
    }
    return errorMessage.join(' - ')
  }

  if (!_message) {
    return <div />
  }

  if (!_.includes(['client', 'server'], type)) {
    console.error('Invalid alert type: ' + type)
    return <div />
  }

  if (!_.includes(Object.keys(errorTypes), status)) {
    console.error('Invalid alert status: ' + status)
    return <div />
  }

  if (!_.isEmpty(error)) {
    _message += ': ' + printError(error!)
  }

  const _fixed: boolean = _.isNil(fixed) ? type === 'client' : fixed
  return (
    <AlertDiv
      className={classNames('type-' + type, 'status-' + status, className, { fixed: _fixed })}
      type={errorTypes[status]}
    >
      {_message}
      {_(onClose).isFunction() ? <FilledRemoveCircle onClick={onCloseIconClicked} /> : undefined}
    </AlertDiv>
  )
}

Alert.propTypes = {
  className: PT.string,
  //  error: PT.oneOfType([AlertErrorPropType, PT.string]),
  fixed: PT.bool,
  message: PT.oneOfType([PT.string, PT.element]),
  onClose: PT.func,
  status: PT.oneOf(['OK', 'ERROR', 'WARNING']),
  type: PT.oneOf(['client', 'server'])
}

Alert.displayName = 'Alert'
export default Alert
