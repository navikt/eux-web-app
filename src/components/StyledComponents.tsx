import styled from 'styled-components'
import {
  fadeIn,
  fadeOut, PaddedHorizontallyDiv
} from '@navikt/hoykontrast'
import { Panel } from '@navikt/ds-react'

export const FadingLineSeparator = styled.div`
   border-left-width: 1px;
   border-left-style: solid;
   border-image: linear-gradient(
    to bottom,
     var(--navds-semantic-color-canvas-background) 0%,
     var(--navds-semantic-color-border) 5%,
     var(--navds-semantic-color-border) 95%,
     var(--navds-semantic-color-canvas-background) 100%
    ) 1 100%;
   opacity: 0;
   &.fadeIn {
     opacity: 0;
     animation: ${fadeIn} 1s forwards;
   }
   &.fadeOut {
     opacity: 1;
     animation: ${fadeOut} 1s forwards;
   }
`
export const HorizontalLineSeparator = styled.div<{size?: string}>`
  height: 1px;
  margin-top: ${(props: any) => props.size || 0}rem;
  margin-bottom: ${(props: any) => props.size || 0}rem;
  background: linear-gradient(90deg,
    var(--navds-semantic-color-canvas-background) 0%,
    var(--navds-panel-color-border) 5%,
    var(--navds-panel-color-border) 95%,
    var(--navds-semantic-color-canvas-background) 100%
  );
  width: 100%;
`

export const Hr = styled.div`
   background: var(--navds-panel-color-border);
   width: 100%;
   height: 1px;
`

export const SpacedHr = styled(Hr)`
   margin-top: 0.5rem;
   margin-bottom: 0.5rem;
`

export const SideBarDiv = styled.div`
  display: flex;
  padding-top: 1.5rem;
`
export const TextAreaDiv = styled.div`
  textarea {
    width: 100%;
  }
`

export const WithErrorPanel = styled(Panel)`
  padding: 0rem;
  background-color: transparent;
  border: none;
  &.error {
    margin: -4px;
    border: 4px solid var(--navds-error-summary-color-border) !important;
  }
`

export const ShadowPanel = styled(Panel)`
 background-color: var(--navds-semantic-color-component-background-alternate);
`

export const AlertstripeDiv = styled.div`
  margin: 0.5rem;
  min-width: 50%;
`

export const Dd = styled.dd`
  width: 60%;
  padding-bottom: 0.35rem;
  padding-top: 0.35rem;
  margin-bottom: 0;
  margin-inline-start: 0;
`
export const Dt = styled.dt`
  width: 40%;
  padding-bottom: 0.35rem;
  padding-top: 0.35rem;
  overflow: hidden;
  text-overflow: ellipsis;
  .typo-element {
    margin-left: 0.5rem;
  }
  font-weight: bold;
`
export const Dl = styled.dl`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  .odd {
    background-color: var(--navds-semantic-color-component-background-alternate);
  }
`
export const RepeatableRow = styled(PaddedHorizontallyDiv)`
  padding-top: 0.3rem;
  padding-bottom: 0.3rem;
  min-height: 3.6rem;
  &.new {
    background-color: rgba(236, 243, 153, 0.5);
  };
  &.error {
    background-color: rgba(255, 0, 0, 0.2);
  };
  &:hover:not(.new):not(.error) {
    background-color: var(--navds-global-color-gray-100);
  }
  &:not(:hover) .control-buttons {
    position: absolute;
    margin-left: -10000px;
  }
`
export const GrayPanel = styled(Panel)`
  background-color: var(--navds-semantic-color-canvas-background);
`
export const TransparentPanel = styled(Panel)`
  background-color: transparent;
`
export const CustomLabel = (props: any) => (<label className='navds-text-field__label navds-label'>{props.children}</label>)
