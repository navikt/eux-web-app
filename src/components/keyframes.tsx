import { keyframes } from 'styled-components'

export const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`
export const animationOpen = keyframes`
  0% {
    height: 0%;
    max-height: 0;
  }
  100% {
    max-height: 150em;
    height: 100%;
  }
`
export const animationClose = keyframes`
  0% {
    max-height: 150em;
    height: 100%;
  }
  100% {
    max-height: 0;
    height: 0%;
  }
`
