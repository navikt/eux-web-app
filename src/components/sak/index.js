import ArbeidsforholdController from './arbeidsforholdController';

export { ArbeidsforholdController };
