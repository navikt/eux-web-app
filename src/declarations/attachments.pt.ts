import { JoarkBrowserType } from 'applications/Vedlegg/JoarkBrowser/JoarkBrowser'
import PT from 'prop-types'

export const JoarkFileVariantPropType = PT.shape({
  variantformat: PT.string.isRequired,
  filnavn: PT.string.isRequired
})

export const JoarkDocPropType = PT.shape({
  dokumentInfoId: PT.string.isRequired,
  tittel: PT.string.isRequired,
  dokumentvarianter: PT.arrayOf(JoarkFileVariantPropType).isRequired
})

export const JoarkPosterPropType = PT.shape({
  journalpostId: PT.string.isRequired,
  tittel: PT.string.isRequired,
  tema: PT.string.isRequired,
  datoOpprettet: PT.instanceOf(Date).isRequired,
  tilleggsopplysninger: PT.arrayOf(PT.string),
  dokumenter: PT.arrayOf(JoarkDocPropType).isRequired
})

export const JoarkFilePropType = PT.shape({
  journalpostId: PT.string.isRequired,
  dokumentInfoId: PT.string.isRequired,
  variant: JoarkFileVariantPropType.isRequired,
  title: PT.string.isRequired,
  tema: PT.string.isRequired,
  date: PT.instanceOf(Date).isRequired
})

export const JoarkFilesPropType = PT.arrayOf(JoarkFilePropType.isRequired)

const ContentPropType = PT.shape({
  base64: PT.string.isRequired
})

export const JoarkBrowserItemWithContentPropType = PT.shape({
  dokumentInfoId: PT.string.isRequired,
  journalpostId: PT.string.isRequired,
  variant: JoarkFileVariantPropType.isRequired,
  title: PT.string.isRequired,
  tema: PT.string.isRequired,
  date: PT.instanceOf(Date).isRequired,

  content: ContentPropType.isRequired,
  name: PT.string.isRequired,
  size: PT.number.isRequired,
  mimetype: PT.string.isRequired
})

export const JoarkBrowserItemFileType = PT.shape({
  key: PT.string.isRequired,
  hasSubrows: PT.bool.isRequired,
  type: PT.oneOf<JoarkBrowserType>(['joark']).isRequired,
  journalpostId: PT.string.isRequired,
  dokumentInfoId: PT.string.isRequired,
  variant: JoarkFileVariantPropType.isRequired,
  title: PT.string.isRequired,
  tema: PT.string.isRequired,
  date: PT.instanceOf(Date).isRequired
})

export const JoarkBrowserItemsFileType = PT.arrayOf(JoarkBrowserItemFileType.isRequired)

export const SEDAttachmentPayloadPropType = PT.shape({
  aktoerId: PT.string.isRequired,
  rinaId: PT.string.isRequired,
  rinaDokumentId: PT.string.isRequired
})
