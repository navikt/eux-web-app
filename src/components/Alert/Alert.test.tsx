import { AlertStatus } from 'declarations/components'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import Alert, { AlertProps } from './Alert'

describe('components/Alert/Alert', () => {
  let wrapper: ReactWrapper
  const initialMockProps: AlertProps = {
    status: 'OK',
    message: 'mockErrorMessage',
    error: undefined,
    onClose: jest.fn()
  }

  it('Render: match snapshot', () => {
    wrapper = mount(<Alert {...initialMockProps} type='server' />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: has proper HTML structure as server', () => {
    wrapper = mount(<Alert {...initialMockProps} type='server' />)
    expect(wrapper.exists('.type-server')).toBeTruthy()
    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockErrorMessage')
  })

  it('Render: has proper HTML structure as client', () => {
    wrapper = mount(<Alert {...initialMockProps} type='client' />)
    expect(wrapper.exists('.type-client')).toBeTruthy()
    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockErrorMessage')
  })

  it('Render: has proper HTML structure with error message', () => {
    wrapper = mount(<Alert {...initialMockProps} type='client' error='mockError' />)
    expect(wrapper.exists('.type-client')).toBeTruthy()
    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockErrorMessage: mockError')
  })

  it('Render: has proper HTML structure as client in OK type', () => {
    wrapper = mount(<Alert {...initialMockProps} type='client' />)
    expect(wrapper.render().hasClass('alertstripe--suksess')).toBeTruthy()
  })

  it('Render: has proper HTML structure as client in WARNING type', () => {
    wrapper = mount(<Alert {...initialMockProps} type='client' status='WARNING' />)
    expect(wrapper.render().hasClass('alertstripe--advarsel')).toBeTruthy()
  })

  it('Render: has proper HTML structure as client in ERROR type', () => {
    wrapper = mount(<Alert {...initialMockProps} type='client' status='ERROR' />)
    expect(wrapper.render().hasClass('alertstripe--feil')).toBeTruthy()
  })

  it('Render: Pretty prints a error message', () => {
    const error = {
      status: 'ERROR' as AlertStatus,
      message: 'message',
      error: 'error',
      uuid: 'uuid'
    }
    wrapper = mount(<Alert {...initialMockProps} type='server' error={error} />)
    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockErrorMessage: message - error - uuid')
  })

  it('Render: Pretty prints a string error', () => {
    const error = 'error'
    wrapper = mount(<Alert {...initialMockProps} type='client' error={error} />)
    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockErrorMessage: error')
  })

  it('Handling: close button clears alert', () => {
    (initialMockProps.onClose as jest.Mock).mockReset()
    wrapper = mount(<Alert {...initialMockProps} type='client' status='ERROR' />)
    wrapper.find('[data-test-id=\'alert__close-icon\']').hostNodes().simulate('click')
    expect(initialMockProps.onClose).toHaveBeenCalled()
  })
})
