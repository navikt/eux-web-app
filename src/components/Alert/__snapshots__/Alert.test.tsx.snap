// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Alert/Alert Render: match snapshot 1`] = `
.c0 {
  opacity: 0;
  -webkit-animation: cJoqxJ 0.5s forwards;
  animation: cJoqxJ 0.5s forwards;
  position: relative;
}

.c0 .alertstripe__tekst {
  -webkit-flex: auto !important;
  -ms-flex: auto !important;
  flex: auto !important;
  margin: auto 0;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .fixed {
  position: fixed;
  top: 0.25rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 10;
}

.c0 .closeIcon {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

.c0 .type-server {
  border-radius: 0px !important;
  border: 0px !important;
}

.c0 .type-server .alertstripe__tekst {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: none !important;
}

.c1 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

<Alert
  message="mockErrorMessage"
  onClose={[MockFunction]}
  status="OK"
  type="server"
>
  <Styled(AlertStripe)
    className="type-server status-OK"
    role="alert"
    type="suksess"
  >
    <AlertStripe
      className="c0 type-server status-OK"
      form="default"
      role="alert"
      size="1.5em"
      type="suksess"
    >
      <div
        className="alertstripe c0 type-server status-OK alertstripe--suksess"
        role="alert"
      >
        <span
          className="alertstripe__ikon"
        >
          <span
            className="sr-only"
          >
            suksess
          </span>
          <Icon
            kind="ok-sirkel-fyll"
            size="1.5em"
          >
            <svg
              focusable="false"
              height="1.5em"
              kind="ok-sirkel-fyll"
              viewBox="0 0 24 24"
              width="1.5em"
            >
              <g
                fill="none"
                fillRule="nonzero"
              >
                <path
                  d="M12 0C5.383 0 0 5.384 0 12s5.383 12 12 12c6.616 0 12-5.384 12-12S18.616 0 12 0z"
                  fill="#1C6937"
                />
                <path
                  d="M9.64 14.441l6.46-5.839a.997.997 0 0 1 1.376.044.923.923 0 0 1-.046 1.334l-7.15 6.464a.993.993 0 0 1-.662.252.992.992 0 0 1-.69-.276l-2.382-2.308a.923.923 0 0 1 0-1.334.997.997 0 0 1 1.377 0l1.717 1.663z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </Icon>
        </span>
        <Normaltekst
          className="alertstripe__tekst"
          tag="div"
        >
          <TypografiBase
            className="alertstripe__tekst"
            tag="div"
            type="normaltekst"
          >
            <div
              className="typo-normal alertstripe__tekst"
            >
              mockErrorMessage
              <Styled(_default)
                data-test-id="c-alert__close-icon"
                onClick={[Function]}
              >
                <_default
                  className="c1"
                  data-test-id="c-alert__close-icon"
                  onClick={[Function]}
                >
                  <svg
                    className="c1"
                    contentScriptType="text/ecmascript"
                    contentStyleType="text/css"
                    data-test-id="c-alert__close-icon"
                    enableBackground="new 0 0 24 24"
                    height={24}
                    id="Filled_Version"
                    onClick={[Function]}
                    preserveAspectRatio="xMidYMid meet"
                    version="1.1"
                    viewBox="0 0 24 24"
                    width={24}
                    x="0px"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0px"
                    zoomAndPan="magnify"
                  >
                    <path
                      d="M11.499,0C5.17,0,0.012,5.148,0,11.477c-0.006,3.072,1.184,5.962,3.352,8.139c2.168,2.176,5.054,3.378,8.126,3.384H11.5  c6.328,0,11.487-5.149,11.5-11.479C23.012,5.181,17.863,0.012,11.499,0z M11.5,22.5L11.5,22.5L11.5,22.5L11.5,22.5z M16.088,15.398  c0.194,0.195,0.193,0.512-0.002,0.707c-0.098,0.098-0.225,0.146-0.353,0.146c-0.127,0-0.256-0.049-0.354-0.146l-3.882-3.898  l-3.896,3.882c-0.098,0.098-0.226,0.146-0.354,0.146c-0.128,0-0.255-0.049-0.353-0.146c-0.195-0.196-0.195-0.513,0.001-0.707  l3.897-3.882L6.91,7.602C6.715,7.406,6.716,7.09,6.912,6.896C7.107,6.699,7.424,6.7,7.619,6.897l3.881,3.897l3.896-3.881  c0.195-0.195,0.513-0.195,0.707,0.001c0.195,0.195,0.195,0.512-0.001,0.707l-3.896,3.882L16.088,15.398z"
                      fill="#3e3832"
                    />
                  </svg>
                </_default>
              </Styled(_default)>
            </div>
          </TypografiBase>
        </Normaltekst>
      </div>
    </AlertStripe>
  </Styled(AlertStripe)>
</Alert>
`;
